service/src/main/kotlin/app/cash/backfila/dashboard/CancelBackfillAction.kt

```diff
@@ -0,0 +1,63 @@
+package app.cash.backfila.dashboard
+
+import app.cash.backfila.service.listener.BackfillRunListener
+import app.cash.backfila.service.persistence.BackfilaDb
+import app.cash.backfila.service.persistence.BackfillState
+import app.cash.backfila.service.persistence.DbBackfillRun
+import app.cash.backfila.service.persistence.DbEventLog
+import javax.inject.Inject
+import misk.MiskCaller
+import misk.exceptions.BadRequestException
+import misk.hibernate.Id
+import misk.hibernate.Query
+import misk.hibernate.Transacter
+import misk.hibernate.load
+import misk.scope.ActionScoped
+import misk.security.authz.Authenticated
+import misk.web.PathParam
+import misk.web.Post
+import misk.web.RequestContentType
+import misk.web.ResponseContentType
+import misk.web.actions.WebAction
+import misk.web.mediatype.MediaTypes
+
+class CancelBackfillAction @Inject constructor(
+  @BackfilaDb private val transacter: Transacter,
+  private val queryFactory: Query.Factory,
+  private val backfillRunListeners: Set<BackfillRunListener>,
+  private val caller: @JvmSuppressWildcards ActionScoped<MiskCaller?>,
+) : WebAction {
+  @Post("/backfill/cancel/{id}")
+  @RequestContentType(MediaTypes.APPLICATION_JSON)
+  @ResponseContentType(MediaTypes.APPLICATION_JSON)
+  @Authenticated
```

Comment:
```suggestion
  @Authenticated(capabilities = ["users"])
```

Empty authenticated will fail in real environments, always set capabilities or services.